.github/workflows/pre-commit.yaml

```diff
@@ -0,0 +1,47 @@
+name: pre-commit
+
+on:
+  push:
```

Comment:
If it's a pre-commit, should it be happening on push?

Reply:
the semantics here is misleading - it's a tool designed to be used locally, using the pre-commit hooks in your .git, it makes most sense (to me) to run it just like any other tool in CI

Reply:
Yep. The idea here is to check that the commit did indeed use pre-commit hooks.

Since users can decide not to install the tool/hooks, we need to check that the commit is consistent with what we expect. Otherwise, in this particular case, we might miss a PR with a lot of trailing white spaces. In such cases, if the PR is merged then fixes will later appear in un-related PRs from people using pre-commit hooks, making review more confusing.